src/simulations/optimisation.py

```diff
@@ -137,17 +149,60 @@ def _power_flow_equations(self, p, netp, grid, hourly_line_losses_pu):
             'undervoltage_costs', (self.grd['n_buses'] - 1, self.N), vtype='continuous'
         )
 
-        # active and reactive loads: netp and netq from kW to W (*1000) to per unit system (/Ab)
-        p.add_list_of_constraints(
-            [
-                pi[:, t] == self.grd['flex_buses'] * netp[:, t] * 1000 / self.grd['base_power']
-                for t in range(self.N)
-            ]
-        )
-        p.add_list_of_constraints(
-            qi[:, t] == self.grd['flex_buses'] * netq[:, t] * 1000 / self.grd['base_power']
-            for t in range(self.N)
-        )
+        # active and reactive loads: modify loads from kW to W (*1000) to per unit system (/Ab)
+        if self.n_homesP > 0:
+            p.add_list_of_constraints(
+                [pi[:, t] == self.grd['flex_buses'] * netp[:, t] * 1000 / self.grd['base_power']
```

Comment:
could you please create a variable for `1000 / self.grd['base_power']`? which a self explanatory name that explains what it is, since it keeps being used.

Reply:
This is done, see `self.kW_to_per_unit_conversion` and `self.per_unit_to_kW_conversion`. Thanks for the comment.

Reply:
wonderful thank you!